mmaction/datasets/pipelines/augmentations.py

```diff
@@ -42,6 +50,166 @@ def _init_lazy_if_proper(results, lazy):
+            to create imgaug augmenter.
+            1) string: only support one string `default`
+            2) list[dict]: create a list of Augmenters by a list of dicts.
+                Each dict contains a key named `type`.
+            3) iaa.Augmenter: Directly use imgaug class.
+    """
+
+    def __init__(self, transforms):
+        if imgaug is None or iaa is None:
+            raise RuntimeError('imgaug is not installed')
+
+        if transforms == 'default':
+            self.transforms = self._default_transforms()
+        elif isinstance(transforms, list):
+            self.transforms = transforms
+        elif isinstance(transforms, iaa.Augmenter):
+            self.aug = self.transforms = transforms
+        else:
+            raise ValueError("transforms must be 'default' or a list of dicts")
+
+        if not isinstance(transforms, iaa.Augmenter):
+            self.aug = iaa.Sequential(
+                [self.imgaug_builder(t) for t in self.transforms])
+
+    def _default_transforms(self):
+        """Default transforms for imgaug."""
+
+        return [
+            dict(type='Rotate', rotate=(-30, 30)),
+            dict(type='Add', value=(-10, 10), per_channel=0.5),
+            dict(
+                type='SomeOf',
+                n=(0, 3),
+                children=[
+                    dict(
+                        type='OneOf',
+                        children=[
+                            dict(type='GaussianBlur', sigma=(0, 0.5)),
+                            dict(type='AverageBlur', k=(2, 7)),
+                            dict(type='MedianBlur', k=(3, 11))
+                        ]),
+                    dict(
+                        type='OneOf',
+                        children=[
+                            dict(
+                                type='Dropout', p=(0.01, 0.1),
+                                per_channel=0.5),
+                            dict(
+                                type='CoarseDropout',
+                                p=(0.03, 0.15),
+                                size_percent=(0.02, 0.05),
+                                per_channel=0.2),
+                        ]),
+                    dict(
+                        type='AdditiveGaussianNoise',
+                        loc=0,
+                        scale=(0.0, 0.05 * 255),
+                        per_channel=0.5),
+                ]),
+        ]
+
+    def imgaug_builder(self, cfg):
+        """Import a module from imgaug.
+
+        It inherits some of :func:`build_from_cfg` logic.
+
+        Args:
+            cfg (dict): Config dict. It should at least contain the key "type".
+
+        Returns:
+            obj: The constructed object.
+        """
+        import inspect
+
+        assert isinstance(cfg, dict) and 'type' in cfg
+        args = cfg.copy()
+
+        obj_type = args.pop('type')
+        if mmcv.is_str(obj_type):
+            obj_cls = getattr(iaa, obj_type)
+        elif inspect.isclass(obj_type):
```

Comment:
Is it possible for users to use this 'instantiation by cls' by simply modify the config file? Maybe add a new tutorial for this.

Reply:
I'll take a look at imgaug source code to find out.

Reply:
```
pipeline = [
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=16,
    ),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(
        type='MultiScaleCrop',
        input_size=224,
        scales=(1, 0.875, 0.75, 0.66),
        random_crop=False,
        max_wh_scale_gap=1,
        num_fixed_crops=13),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(type='Flip', flip_ratio=0.5),
    dict(type='Imgaug', transforms='default'),
    # dict(type='Imgaug', transforms=[dict(type='Rotate', rotate=(-30, 30))]),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
```
Maybe show a way to build it from config such as this. Seems that you need to import iaa in the config file.

Reply:
I misunderstood you last night... Add some tutorials in class comments.

If `type` is iaa.Augmenter subclass, we need to import iaa first. If we use string `type`, there is no need to import iaa.